yext/search-ui-react
# remove react-router-dom

remove any usage from brower history state with react-router-dom library.

Previously, we rely on external page setup hook to execute search when a new path is push to browser state (from useHistory hook in react-router-dom). Update spellcheck, verticalLink, and searchbar executeQuery function to support user-defined behavior if provided (e.g. if user want to switch to another page using react router). Otherwise, they should just invoke a search by default.

fix bug in useSearchWithNearMeHandling where executeQuery async function is not getting the latest query and verticalKey. Move those answersActions statements into the function solve the issue.

J=SLAP-1892
TEST=manual

test with starter repo:
- click on vertical link without onClickVerticalLink defined, see that a search is executed for universal search page. 
- click on vertical link with onClickVerticalLink defined with a new browser path to another vertical page, see that url is updated and a search is executed as expected for vertical search page. 
- click on spell check with/without onClick defined, see that a search is executed for the page.
- click on an autocomplete suggestion from searchbar, see that a search is perform with/without onSearch function
